File: GTP2/gtp2.cpp
#include<iostream>

using namespace std;

// void fill(int n[], int b, int l) {
//         for(int i = 0; i < l; i++) {
//                 n[i] = b*(i+1);
//         }
// }

void fill(int n[], int b, int l) {
        if(l > 0) {
                n[l-1] = b*(l);
                fill(n, b, l-1);
        }
}

void print(int n[], int l) {
        for(int i = 0; i < l; i++) {
                cout << n[i] << " ";
        }

        cout << "\n";
}

int main() {
        int l, b;

        cout << "Ingrese el tamaño del arreglo: ";
        cin >> l;

        cout << "Ingrese la base: ";
        cin >> b;

        int numbers[l];

        fill(numbers, b, l);

        print(numbers, l);

        return 0;
}

File: GTP2/gtp2.py
def get_multiplos(SIZE:int, base:int) -> list[int]:
        return [base*i for i in range(1, SIZE+1)]

def print_vect(v:list[int]):
        for i in v: print(f"{i}", end=" ")
        print()

def main():
        SIZE:int; base:int
        v:list[int] = []

        print("Ingrese la dimensión del arreglo: ")
        SIZE = int(input("> "))

        print("Ingrese la base: ")
        base = int(input("> "))

        v = get_multiplos(SIZE, base)

        print_vect(v)     

if __name__ == "__main__":
        main()
